spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -0,0 +1,1380 @@
+    /// The `data` field contains a `ScalaBuildTarget` object
+    SCALA = "scala"
+    /// The `data` field contains a `SbtBuildTarget` object.
+    SBT = "sbt"
+}
+
+structure BuildTargetCapabilities {
+    /// This target can be compiled by the BSP server.
+    @required
+    canCompile: Boolean
+    /// This target can be tested by the BSP server.
+    @required
+    canTest: Boolean
+    /// This target can be run by the BSP server.
+    @required
+    canRun: Boolean
+    /// This target can be debugged by the BSP server.
+    @required
+    canDebug: Boolean
+}
+
+@enumKind("open")
+@dataFieldDiscriminator
+enum BuildTargetTag {
+    /// Target contains re-usable functionality for downstream targets. May have any
+    /// combination of capabilities.
+    LIBRARY = "library"
+    /// Target contains source code for producing any kind of application, may have
+    /// but does not require the `canRun` capability.
+    APPLICATION = "application"
+    /// Target contains source code for testing purposes, may have but does not
+    /// require the `canTest` capability.
+    TEST = "test"
+    /// Target contains source code for integration testing purposes, may have
+    /// but does not require the `canTest` capability.
+    /// The difference between "test" and "integration-test" is that
+    /// integration tests traditionally run slower compared to normal tests
+    /// and require more computing resources to execute.
+    INTEGRATION_TEST = "integration-test"
+    /// Target contains source code to measure performance of a program, may have
+    /// but does not require the `canRun` build target capability.
+    BENCHMARK = "benchmark"
+    /// Target should be ignored by IDEs.
+    NO_IDE = "no-ide"
+    /// Actions on the target such as build and test should only be invoked manually
+    /// and explicitly. For example, triggering a build on all targets in the workspace
+    /// should by default not include this target.
+    ///
+    /// The original motivation to add the "manual" tag comes from a similar functionality
+    /// that exists in Bazel, where targets with this tag have to be specified explicitly
+    /// on the command line.
+    ///
+    MANUAL = "manual"
+}
+
+list BuildTargetTags {
+    member: BuildTargetTag
+}
+
+/// The Task Id allows clients to uniquely identify a BSP task and establish a client-parent relationship with another task id.
+structure TaskId {
+    /// A unique identifier
+    id: Identifier
+    // The parent task ids, if any. A non-empty parents field means
+    // this task is a sub-task of every parent task id. The child-parent
+    // relationship of tasks makes it possible to render tasks in
+    // a tree-like user interface or inspect what caused a certain task
+    // execution.
+    parents: Identifiers
+}
+
+string Identifier
+
+list Identifiers {
+    member: Identifier
+}
+
+/// Included in notifications of tasks or requests to signal the completion state.
+@enumKind("open")
```

Comment:
Same comment as above, I'm inclined to say this should be `closed`. What are you thoughts about keeping it `open`?

Reply:
The fact that is open is accurate as per the current specification. 

I cannot comment as to whether this is correct or not 🤷 From an open-protocol standpoint, which BSP aims at being, as far as I follow, it makes sense to accept any int value. For instance, you could have a BSP client and a BSP server that would have an understanding of `4` representing "DELAYED". If you go from open to close, it means you're preventing that. 

**However**, you can have bespoke code-generators that would generate open-enumerations as closed ones 🤷‍♂️. In Scala, the difference is basically between :  

```scala 
enum StatusCode {
   case Ok 
   case Error
   case Cancelled 
} 
``` 

and 

```scala 
enum StatusCode {
   case Ok 
   case Error
   case Cancelled
   case Other(value: Int) 
} 
```

Reply:
Alright, makes sense 👍🏼